File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int curr = 1e7;
        for(int i = 0; i < prices.size(); ++i){
            if(prices[i] < curr)
                curr = prices[i];
            else
                ans += prices[i] - curr, curr = prices[i];
        }
        return ans;
    }
};

File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mn, mx, ans = prices[0], prices[0], 0
        for price in prices:
            if price < mx:
                ans += (mx - mn)
                mx = mn = price
            elif price < mn:
                mn = mx = price
            elif price > mx:
                mx = price
        ans += (mx - mn)
        return ans